firebase-vscode/src/data-connect/service.ts

```diff
@@ -243,23 +234,23 @@ export class DataConnectService {
       extensions: this._auth(),
     });
     if (params.instance === InstanceType.PRODUCTION) {
-      const resp = await executeGraphQL(servicePath, prodBody);
+      const client = dataconnectDataplaneClient();
+      const resp = await executeGraphQL(client, servicePath, prodBody);
       return this.handleProdResponse(resp);
     } else {
-      const resp = await fetch(
-        (await this.emulatorsController.getLocalEndpoint()) +
-          `/v1beta/${servicePath}:executeGraphql`,
-        {
-          method: "POST",
-          headers: {
-            Accept: "application/json",
-            "Content-Type": "application/json",
-            "x-mantle-admin": "all",
-          },
-          body,
-        },
-      );
-      return this.handleResponse(resp);
+      const endpoint = this.emulatorsController.getLocalEndpoint();
+      if (!endpoint) {
+        throw new DataConnectError(
+          `Emulator isn't running. Please start your emulator!`,
+        );
+      }
+      const client = new Client({
+        urlPrefix: endpoint,
+        apiVersion: "v1beta",
```

Comment:
Probably should use DATACONNECT_API_VERSION here

Reply:
Done